### ISSUE TITLE
Error Upgrading Nx 7 app to Nx 8 - Cannot find workspace.json 

### ISSUE THREAD

**Issue by frankwb24** [nrwl/nx#1684]
I'm getting a "Cannot find workspace.json" error upgrading an app that is using Nx 7.7.1. I was able to reproduce the error just creating a new app using nx 7.7.1 and doing an update.

Steps to reproduce:

npm install -g @nrwl/schematics@7.7.1
create-nx-workspace test
cd test
npm run update

**Comment by rafa-as** [comment]
Same here

**Comment by rafa-as** [comment]
@frankwb24 you can use one workaround: copy the angular.json fil, name it as workspace.json and  execute npm run update

**Comment by frankwb24** [comment]
@rafa-as Yes, that worked. Thanks.

**Comment by rafa-as** [comment]
@vsavkin @brandonroberts what do you think?

**Comment by rafa-as** [comment]
@vsavkin I found this function that maybe you should call to get the workspace file on every place:

```typescript
export function readWorkspaceConfigPath(): any {
  if (fileExists('workspace.json')) {
    return readJsonFile('workspace.json');
  } else {
    return readJsonFile('workspace.json');
  }
}
```
Obviously fixing the else with angular.json instead workspace.json. 

Also I found another related thing. In the update process, when tries to update the angular cli this message appears:

Updating @angular/cli to 8.0.1
`"Package not installed: "--allow-dirty". Skipping."`

Seems to be something about:

```typescript
const args = [
            'update',
            `${options.package}@${options.to}`,
            '--force',
            '--allow-dirty'
```

**Comment by vsavkin** [comment]
Hey folks. Thank you for troubleshooting.

I released 8.4.4 with a fix. You should be able to update from 7x to 8.4.4+ without any issues.

`allow-dirty` allows us to work around some limitations of `ng update`.

**Comment by rafa-as** [comment]
Thanks @vsavkin !

About the allow dirty, I think that it's interpreted as a package instead of a command parameter.

You can see the piece of code that throws the warning:

``` typescript
    const version = projectDeps.get(npmName); 
    if (!version) {
      logger.warn(`Package not installed: ${JSON.stringify(npmName)}. Skipping.`);
      continue;
    }
```
